packages/components/src/presentation-components/properties/NavigationPropertyTargetSelector.tsx

```diff
@@ -144,22 +137,91 @@ function getNavigationTargetFromPropertyRecord(record: PropertyRecord): Navigati
   return { key: value.value as InstanceKey, label: LabelDefinition.fromLabelString(value.displayValue) };
 }
 
-function TargetSelectControl<TOption, IsMulti extends boolean = boolean>({ children, ...props }: ControlProps<TOption, IsMulti>) {
+function TargetSelectControl(props: ControlProps<NavigationPropertyTarget, false>) {
+  const { hasValue, getValue, selectProps } = props;
+  const [inputValue, setInputValue] = useState<string>(() => getValue()[0]?.label.displayValue ?? "");
+  const inputRef = useRef<HTMLInputElement>(null);
+  const selectedValue = getValue()[0];
+
+  useEffect(() => {
+    // istanbul ignore else
+    if (hasValue && selectedValue) {
+      setInputValue(selectedValue.label.displayValue);
+    }
+  }, [hasValue, selectedValue]);
+
+  const handleMenuOpen = () => {
+    selectProps.onMenuOpen();
+    inputRef.current?.focus();
+    inputRef.current?.toggleAttribute
+    selectProps.onInputChange("", { action: "input-change", prevInputValue: inputValue });
+  }
+
+  const handleInputBlur = () => {
+    if (selectedValue?.label.displayValue) {
+      setInputValue(selectedValue.label.displayValue);
+      selectProps.onInputChange(selectedValue.label.displayValue, { action: "set-value", prevInputValue: inputValue });
+    }
+    else {
+      setInputValue("");
+      selectProps.onInputChange("", { action: "set-value", prevInputValue: inputValue });
+    }
+    selectProps.onMenuClose();
+  }
+
+  const handleDropdownButtonClick = () => {
+    if (selectProps.menuIsOpen) {
+      selectProps.onMenuClose();
+      inputRef.current?.blur();
+    }
+    else {
+      handleMenuOpen();
+    }
+  }
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    selectProps.onMenuOpen();
+    selectProps.onInputChange(event.target.value, { action: "input-change", prevInputValue: inputValue });
+  }
+
+  /** This function is used to cancel overriden react-select keyboard events. */
+  const handleInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.code === "Space" || event.code === "Home" || event.code === "End") {
+      event.stopPropagation();
+    }
+    if (event.key === "Tab") {
+      selectProps.onMenuClose();
+    }
+  }
+
   return (
     <components.Control {...props} className="iui-input-with-icon">
-      {children}
+      <components.ValueContainer {...props} className="iui-select-button" innerProps={{ onClick: handleMenuOpen, style: { cursor: "text" } }}>
+        <Input
+          ref={inputRef}
+          type="search"
+          value={inputValue}
+          onBlur={handleInputBlur}
+          onFocus={handleMenuOpen}
+          onChange={handleInputChange}
+          style={{ width: "100%", height: "100%", border: "none", textOverflow: "ellipsis", overflow: "hidden", outline: "none", padding: "0" }}
```

Comment:
I thinks these styles should be moved into separate `scss` file.

Reply:
But I want to override itwinui styles, should I mark them as !important in the `scss` file?

Reply:
You can add some class on input parent container?